### DISCUSSION TITLE
[Bug] Shrink sidebar until hovered of userChrome.css after FF 133

### DISCUSSION THREAD

**Discussion by marrak** [piroor/treestyletab#3669] [Q&A] [answered]
## Abstract
After upgrade of Firefox from 132.0.2 to 133.0 Shrink sidebar until hovered does not work
## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Use userChrome.css
 /* Shrink sidebar until hovered, when using Tree Style Tab. */
:root {
    --thin-tab-width: 1px;
    --wide-tab-width: 100px;
}
#sidebar-box:not([sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"]) {
    min-width: var(--wide-tab-width) !important;
    max-width: none !important;
}
#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] {
    position: relative !important;
    transition: all 100ms !important;
    min-width: var(--thin-tab-width) !important;
    max-width: var(--thin-tab-width) !important;
}
#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"]:hover {
    transition: all 200ms !important;
    min-width: var(--wide-tab-width) !important;
    max-width: var(--wide-tab-width) !important;
    margin-left: calc((var(--wide-tab-width) - var(--thin-tab-width)) * -1) !important;
}
 

## Expected result
Tree is properly shrinking

## Actual result
Tree is not visible at all (only tool tips of cards can be seen)

## Workaround
Changing
    margin-left: calc((var(--wide-tab-width) - var(--thin-tab-width)) * -1) !important;
to
    margin-left: -1px !important;

alt least shows tab, but works hard :/

## Environment

 * Platform (OS): Win10
 * Version of Firefox: 133.0 
 * Version (or revision) of Tree Style Tab: 4.0.25

**Comment by piroor** [comment]
userChrome.css compatibility issue is Firefox side problem and out of purpose of the issue tracker, so I've moved this to the discussions.

**Comment by seqizz** [comment, marked as answer]
Thanks for the hint, I was very confused.

For me the problem was that the z-index was changed, so even though sidebar was expanding, I was not able to see the parts which was expanding. Adding `z-index: 999 !important;` into `#sidebar-box` section solved it.

**Comment by dlechuga** [comment]
Check out the code snippets from the Wiki, the recommended [Auto hide theme](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#full-auto-showhide-theme) seems to be working fine with Firefox 133.0, it looks like the issue is indeed a change with the `z-index` in the latest browser update.
